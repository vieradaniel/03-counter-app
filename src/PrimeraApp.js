import React from 'react';
import { Fragment } from 'react/cjs/react.development';
import PropTypes from 'prop-types';


const PrimeraApp=({ saludo, subtitulo = "soy un subtitulo" })=>{

  

    return (
        <>
            <h1>{ saludo }</h1>
            <p>{subtitulo}</p> 
        </>
    ); 

}

PrimeraApp.propTypes ={
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps ={
    subtitulo: "soy un subtitulo"
}

export default PrimeraApp;