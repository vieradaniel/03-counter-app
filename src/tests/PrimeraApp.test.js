import React from 'react';

import'@testing-library/jest-dom'
import { shallow, ShallowWrapper } from 'enzyme';
import  PrimeraApp  from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />',()=>{

    test('debe de mostrar <PrimeraApp /> correctamente',()=>{
        const saludo = "hola, soy goku";
        const wrapper = shallow(<PrimeraApp saludo = "hello" />);
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar el subtitulo enviado por props', () => {
        const saludo = "hola, soy goku";
        const subtitulo= "soy un subtitulo";
        const wrapper = shallow(
            <PrimeraApp 
                saludo = {saludo} 
                subtitulo = {subtitulo}
            />
        );

        const textoParrafo = wrapper.find('p').text();
        expect( textoParrafo ).toBe( subtitulo );
    });
    

})