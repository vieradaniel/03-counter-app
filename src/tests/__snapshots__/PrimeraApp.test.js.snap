// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    hello
  </h1>
  <p>
    soy un subtitulo
  </p>
</Fragment>
`;
